packages/kbn-ftr-common-functional-services/services/retry/retry_for_success.ts

```diff
@@ -44,27 +44,45 @@ interface Options<T> {
   onFailureBlock?: () => Promise<T>;
   onFailure?: ReturnType<typeof defaultOnFailure>;
   accept?: (v: T) => boolean;
+  description?: string;
   retryDelay?: number;
+  retryCount?: number;
 }
 
 export async function retryForSuccess<T>(log: ToolingLog, options: Options<T>) {
   const {
+    description,
     timeout,
     methodName,
     block,
     onFailureBlock,
     accept = returnTrue,
     retryDelay = 502,
+    retryCount,
   } = options;
   const { onFailure = defaultOnFailure(methodName) } = options;
 
   const start = Date.now();
   const criticalWebDriverErrors = ['NoSuchSessionError', 'NoSuchWindowError'];
   let lastError;
+  let attemptCounter = 0;
+  const addText = (str: string | undefined) => (str ? ` waiting for '${str}'` : '');
 
   while (true) {
+    if (retryCount) {
+      // Use retryCount as an optional condition
+      if (++attemptCounter > retryCount) {
+        onFailure(
```

Comment:
What happens if `onFailure` doesn't break the while loop? The default implementation throws an error but what if a custom `onFailure` handler doesn't?

Reply:
good catch! I will make a fix. Still you can't pass custom onFailure logic outside Retry service.